qa/common/gen_qa_trt_format_models.py

```diff
@@ -0,0 +1,432 @@
+]
+'''.format(model_name, max_batch, version_policy_str,
+           np_to_model_dtype(input_dtype), tu.shape_to_dims_str(input_shape),
+           np_to_model_dtype(input_dtype), tu.shape_to_dims_str(input_shape),
+           np_to_model_dtype(output0_dtype),
+           tu.shape_to_dims_str(output0_shape),
+           np_to_model_dtype(output1_dtype),
+           tu.shape_to_dims_str(output1_shape))
+
+    try:
+        os.makedirs(config_dir)
+    except OSError as ex:
+        pass  # ignore existing dir
+
+    with open(config_dir + "/config.pbtxt", "w") as cfile:
+        cfile.write(config)
+
+
+def create_plan_model(models_dir, max_batch, model_version, input_shape,
+                      output0_shape, output1_shape, input_dtype, output0_dtype,
+                      output1_dtype, input_memory_format, output_memory_format):
+
+    if not tu.validate_for_trt_model(input_dtype, output0_dtype, output1_dtype,
+                                     input_shape, output0_shape, output1_shape):
+        return
+
+    create_plan_modelconfig(models_dir, max_batch, model_version, input_shape,
+                            output0_shape, output1_shape, input_dtype,
+                            output0_dtype, output1_dtype, input_memory_format,
+                            output_memory_format, None)
+
+    if (not tu.shape_is_fixed(input_shape) or
+            not tu.shape_is_fixed(output0_shape) or
+            not tu.shape_is_fixed(output1_shape)):
+        create_plan_dynamic_modelfile(models_dir, max_batch, model_version,
+                                      input_shape, output0_shape, output1_shape,
+                                      input_dtype, output0_dtype, output1_dtype,
+                                      input_memory_format, output_memory_format)
+    else:
+        create_plan_fixed_modelfile(models_dir, max_batch, model_version,
+                                    input_shape, output0_shape, output1_shape,
+                                    input_dtype, output0_dtype, output1_dtype,
+                                    input_memory_format, output_memory_format)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--models_dir',
+                        type=str,
+                        required=True,
+                        help='Top-level model directory')
+    FLAGS, unparsed = parser.parse_known_args()
+
+    # reformat-free input
+    # Fixed shape
+    create_plan_model(FLAGS.models_dir, 0, 1, (13, 2, 1), (13, 2, 1),
+                      (13, 2, 1), np.float16, np.float16, np.float16,
+                      trt.TensorFormat.CHW2, trt.TensorFormat.LINEAR)
+    create_plan_model(FLAGS.models_dir, 8, 1, (13, 2, 1), (13, 2, 1),
+                      (13, 2, 1), np.float16, np.float16, np.float16,
+                      trt.TensorFormat.CHW2, trt.TensorFormat.LINEAR)
+
+    # Dynamic shape
+    # FIXME The generated model inputs are LINEAR format instead of CHW32 format
```

Comment:
Is this comment still valid? I see L117 and L118 will get trt.TensorFormat.CHW32.

Reply:
Yes, the corresponding test doesn't manipulate the input as the input format is actually linear.